src/Workspaces/Remote/ServiceHub/Shared/RemoteEndPoint.cs

```diff
@@ -194,7 +210,8 @@ private void OnDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
             _debuggingLastDisconnectReason = e;
             _debuggingLastDisconnectCallstack = new StackTrace().ToString();
 
-            if (e.Reason != DisconnectedReason.LocallyDisposed)
+            if (e.Reason != DisconnectedReason.LocallyDisposed &&
+                e.Reason != DisconnectedReason.RemotePartyTerminated)
             {
                 // log when this happens
                 LogDisconnectInfo(e, new StackTrace().ToString());
```

Comment:
should this not just use _debuggingLastDisconnectCallstack ?

Reply:
Yes.